src/smolagents/agents.py

```diff
@@ -1147,13 +1144,13 @@ def __init__(
         grammar: Optional[Dict[str, str]] = None,
         additional_authorized_imports: Optional[List[str]] = None,
         planning_interval: Optional[int] = None,
-        use_e2b_executor: bool = False,
+        executor_type: str = "local",
```

Comment:
You can also consider using enums for higher type safety, e.g:
```py
from enum import Enum

class ExecutorType(Enum):
    E2B = "e2b"
    DOCKER = "docker"
    LOCAL = "local"
```
```suggestion
        executor_type: ExecutorType = ExecutorType.LOCAL,
```